app.go

```diff
@@ -24,142 +24,238 @@ import (
 	"context"
 	"os"
 	"os/signal"
-	"reflect"
 	"syscall"
 	"time"
 
-	"github.com/pkg/errors"
 	"go.uber.org/dig"
 	"go.uber.org/fx/internal/fxlog"
 	"go.uber.org/fx/internal/fxreflect"
 	"go.uber.org/multierr"
 )
 
-// App models a modular application
+// DefaultTimeout is the start and stop timeout used by Run.
+const DefaultTimeout = 15 * time.Second
+
+// Timeout is a convenience function to construct a context with a timeout.
+// It's only intended to reduce noise in the main function; since it doesn't
+// expose context.CancelFunc, it may leak resources.
+func Timeout(d time.Duration) context.Context {
```

Comment:
Opportunity for misuse is high here if someone decides that they can just use this instead of `context.WithTimeout`.

This will be used only if people use `Start` and `Stop` directly at which point they can build their own contexts.